src/Mod/Sketcher/Gui/TaskSketcherElements.h

```diff
@@ -177,6 +177,38 @@ protected Q_SLOTS:
     void onItemHovered(QListWidgetItem *);
 };
 
+class ElementFilterList : public QListWidget
+{
+    Q_OBJECT
+
+public:
+    explicit ElementFilterList(QWidget* parent = nullptr);
+    ~ElementFilterList() override;
+
+protected:
+    void changeEvent(QEvent* e) override;
+    virtual void languageChange();
+
+private:
+    inline static const std::vector<char *> filterItems = {
+        QT_TR_NOOP("Normal"),
+        QT_TR_NOOP("Construction"),
+        QT_TR_NOOP("Internal"),
+        QT_TR_NOOP("External"),
+        QT_TR_NOOP("All types"),
+        QT_TR_NOOP(" - Point"),
+        QT_TR_NOOP(" - Line"),
+        QT_TR_NOOP(" - Circle"),
+        QT_TR_NOOP(" - Ellipse"),
+        QT_TR_NOOP(" - Arc of circle"),
+        QT_TR_NOOP(" - Arc of ellipse"),
+        QT_TR_NOOP(" - Arc of hyperbola"),
+        QT_TR_NOOP(" - Arc of parabola"),
+        QT_TR_NOOP(" - B-Spline")

```

Comment:
These are going to be ugly for translators -- each of those words will already have a translation, this is going to add a whole new set with the added punctuation. I can think of a couple different solutions, depending on how desperate we are for speed (obviously an inline static vector of pointers is about as space- and time-efficient as you can get). Is this coded that way just to avoid "premature pessimization" or is it really performance sensitive?

Reply:
Maybe we can just do something like this : 
`qstring = " - " + QT_TR_NOOP("Ellipse")`
I don't think it's performance sensitive, I just didn't thought about translation.

Reply:
Yes, that's basically how we handle this kind of thing elsewhere, so if you are OK with this being a vector of `QString`s (or a `QStringList` or whatever you want to use instead), then that's how I would approach it.

Reply:
(Don't forget you will have to change the code that handles the language, though, or you will lose the "-" when it happens)

Reply:
@0penBrain what is the best way to do it here? I'm not sure where to add it in your code. Especially so that we don't forget it later as Chennes pointed out.
I tried replacing `QT_TR_NOOP(" - Point")`
by `strcat(" - ", QT_TR_NOOP("Point"))`
But that prevent fc from launching.

Reply:
> @0penBrain what is the best way to do it here?

I just pushed a proposal PR to your repo : https://github.com/PaddleStroke/FreeCAD/pull/5
@chennes would this looks fine to you ?